### PR TITLE
Bump nixpkgs to 22.05 release branch

### PR DESCRIPTION
**Pull request by nomeata**

### CODE REVIEW

**Comment by nomeata** [comment]
First issue: https://github.com/WebAssembly/spec/issues/1480

**Comment by nomeata** [comment]
Also affected by https://github.com/NixOS/nixpkgs/issues/155113

**Comment by nomeata** [comment]
We are moving away from asciidoc and antora anyways, right? So I can just dump the antora-related CI jobs, instead of investing time into getting them to work?

**Comment by nomeata** [comment]
We are moving away from asciidoc and antora anyways, right, @crusso? So I can just dump the antora-related CI jobs, instead of investing time into getting them to work?

**Comment by nomeata** [comment]
Whoohoo, that upgrade seems to go through fine :-)

**Comment by crusso** [comment]
> We are moving away from asciidoc and antora anyways, right, @crusso? So I can just dump the antora-related CI jobs, instead of investing time into getting them to work?

Eventually, but I think I'll still want asciidoctor (not asciidoc) working for a while (thought asciidoctor isn't even used by CI I expect)

**Comment by nomeata** [comment]
Wouldn't we just have .md files once #3172 merged?

**Comment by crusso** [comment]
> Wouldn't we just have .md files once #3172 merged?

Well, no. I like the fact that adoc lets you include files (and regions of) files that we can check separately. Until we have a workaround, I'd prefer to generate the markdown off the adoc.

Next question: Is there a workaround? I guess we could write a script to extract the examples somehow, but yuck.

**Comment by crusso** [comment]
https://github.com/ocaml/omd might let us add a .md mode to moc for checking code embedded in markdown files, but it doesn't handle gfm tables... sigh

**Comment by nomeata** [comment]
I see, so we'll keep the adocs, and generate md for the doc site?

But even then we don't need antora, just any adoc processor, in our CI, right?

**Comment by crusso** [comment]
> I see, so we'll keep the adocs, and generate md for the doc site?
> 
> But even then we don't need antora, just any adoc processor, in our CI, right?

I think that's correct. Indeed, for the conversion I need to use asciidoctor, not asciidoc. Sorry for the delay.

**Comment by nomeata** [comment]
Depends on #3350

File: default.nix [hunk 1]
```diff
@@ -313,7 +313,7 @@ rec {
     };
 
     testDerivationDeps =
-      (with nixpkgs; [ wabt bash perl getconf moreutils nodejs-16_x sources.esm ]) ++
+      (with nixpkgs; [ wabt bash perl getconf moreutils nodejs-16_x ]) ++ # TODO: sources.esm
```
Comment (nomeata) [inline default.nix:316]: Unclear why we needed this, but it seems we don’t need it any more.
```suggestion
      (with nixpkgs; [ wabt bash perl getconf moreutils nodejs-16_x ]) ++
```

**Comment by nomeata** [comment]
Looks good, can go in, I think

**Review by ggreif — APPROVED** [review]
Great!

**Comment by ggreif** [comment]
Looks like `nix/155128.patch` got added, but never applied by `nix`. @nomeata was  this intentional?

**Comment by nomeata** [comment]
Hmm, maybe it wasn't needed after all. Feel free to remove.

**Pull request merged**